packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -394,31 +394,35 @@ export class Pipeline extends PipelineBase {
 
   /** @internal */
   public _attachActionToPipeline(stage: Stage, action: IAction, actionScope: CoreConstruct): FullActionDescriptor {
-    const richAction = new RichAction(action, this);
+    const actionEnv = new ActionEnvironment(action, this);
 
     // handle cross-region actions here
-    const crossRegionInfo = this.ensureReplicationResourcesExistFor(richAction);
+    const crossRegionInfo = this.ensureReplicationResourcesExistFor(actionEnv);
 
     // get the role for the given action, handling if it's cross-account
-    const actionRole = this.getRoleForAction(stage, richAction, actionScope);
-
-    // // CodePipeline Variables
-    validateNamespaceName(richAction.actionProperties.variablesNamespace);
```

Comment:
That's fair. 

Well this is gone for practical reasons: by this point the namespace has turned into a `${Token[]}` and so cannot be validated anymore.

I have no way of fixing that without the Action validating itself.

Reply:
Then the validation needs to be moved into CodePipeline's `validation`.

(It probably should have been there from the start, because of the fact that namespaces and artifact names have to be unique, and we should check that: https://github.com/aws/aws-cdk/issues/6111 ).